rapids/README.md

```diff
@@ -14,21 +14,104 @@ action supports Dask and Spark runtimes for RAPIDS on
 [best practices](/README.md#how-initialization-actions-are-used) of using
 initialization actions in production.
 
-## Spark RAPIDS XGBoost
+## Spark RAPIDS Accelerator
+
+### Prerequisites
+
+To use Spark Rapids SQL plugin, XGBoost4j with Spark 3
+*   Apache Spark 3.0.0 RC1+
+*   Hardware Requirements
+    *   NVIDIA Pascal™ GPU architecture or better (V100, P100, T4 and later)
+    *   Multi-node clusters with homogenous GPU configuration
+*   Software Requirements
+    *   NVIDIA driver 440.33+
+    *   CUDA v10.2/v10.1/v10.0
+    *   NCCL 2.4.7 and later
+
+This section describes how to create
+[Google Cloud Dataproc](https://cloud.google.com/dataproc) cluster with [Spark RAPIDS SQL plugin](https://github.com/NVIDIA/spark-rapids) and [XGBoost4j](https://github.com/rapidsai/spark-examples/tree/support-spark3.0).
+
+### Step 1. Create Dataproc cluster with Spark RAPIDS Accelerator
+
+The following command will create a new Dataproc cluster named `CLUSTER_NAME`
+with installed GPU drivers, Spark RAPIDS Accelerator, Spark RAPIDS XGBoost libraries and Jupyter Notebook.
+
+A few notes:
+
+* spark.yarn.unmanagedAM.enabled=false is set since it current breaks SparkUI, will remove later
+* for better GPU performance we recommend to remove IO bottleneck as much as possible, that includes faster disk/networking.
+* Adjust spark properties in cluster creation command to the hardware availabilty 
+* For best practice, please refer to NVIDIA [getting started guide](https://nvidia.github.io/spark-rapids/)
+
+```bash
+export CLUSTER_NAME=<cluster_name>
+export GCS_BUCKET=<your bucket for the logs and notebooks>
+export REGION=<region>
+export ZONE=<zone>
+export NUM_GPUS=2
+export NUM_WORKERS=2
+
+gcloud beta dataproc clusters create $CLUSTER_NAME  \
+    --zone $ZONE \
+    --region $REGION \
+    --image-version=2.0.0-RC2-ubuntu18 \
+    --master-machine-type n1-standard-4 \
+    --master-boot-disk-size 400 \
+    --master-boot-disk-type pd-ssd \
+    --num-workers $NUM_WORKERS \
+    --worker-accelerator type=nvidia-tesla-t4,count=$NUM_GPUS \
+    --worker-machine-type n1-standard-16 \
+    --worker-boot-disk-size 400 \
+    --master-boot-disk-type pd-ssd \
+    --initialization-actions gs://goog-dataproc-initialization-actions-${REGION}/gpu/install_gpu_driver.sh,gs://goog-dataproc-initialization-actions-${REGION}/rapids/rapids.sh \
+    --optional-components=ANACONDA,JUPYTER,ZEPPELIN \
+    --metadata gpu-driver-provider="NVIDIA" \
+    --metadata rapids-runtime=SPARK \
+    --metadata install-gpu-agent=true \
+    --bucket $GCS_BUCKET \
+    --subnet=default \
+    --enable-component-gateway \
+    --properties="^#^spark:spark.dynamicAllocation.enabled=false#spark:spark.shuffle.service.enabled=false#spark:spark.yarn.unmanagedAM.enabled=false#spark:spark.task.resource.gpu.amount=0.125#spark:spark.executor.cores=8#spark:spark.executor.memory=20G"
+```
```

Comment:
Can we move this into the initialization action as well?

Reply:
I prefer to leave them out for the following reasons.
```
spark:spark.dynamicAllocation.enabled=false#spark:spark.shuffle.service.enabled=false
might be needed for dynamic allocation, I need to experiment on whether this will work and it is more likely to be working after per stage scheduling in enabled after spark 3.1

spark:spark.yarn.unmanagedAM.enabled=false
I am only setting this because it will be break sparkUI in current dataproc rc2 image.

spark:spark.task.resource.gpu.amount=0.125#spark:spark.executor.cores=8#spark:spark.executor.memory=20G"
These config will vary based on your cluster hardware. Should be configured on a per job level.  So optional.
```

Reply:
I know we set dynamic allocation at the init script level for other scripts we have such as [h2o.ai](https://github.com/GoogleCloudDataproc/initialization-actions/blob/5c4e340dc326639fb39733c77f9d42a046486584/h2o/h2o.sh#L51). If this is still a necessary config for Spark 3.0 and RAPIDS I would definitely say we should move it here. We can always remove it later. 

For `spark:spark.yarn.unmanagedAM.enabled=false`, does this specifically apply to when RAPIDS is installed on the cluster? (I can also look into this a bit more)

For `spark:spark.task.resource.gpu.amount=0.125#spark:spark.executor.cores=8#spark:spark.executor.memory=20G`, should we maybe just move this to the job submission part of the README?

Reply:
`spark:spark.yarn.unmanagedAM.enabled=false` moves yarn application master to spark driver, so that we fully utilize the cluster, I am happy to remove it if the sparkUI issue is fixed in the dataproc v2 preview image.

 spark:spark.task.resource.gpu.amount=0.125#spark:spark.executor.cores=8#spark:spark.executor.memory=20G will be usefully for notebook jobs.

Let me experiment on dynamic allocation and see how it works, I am a little concern on whether yarn treat GPU as first class resource as the moment.